(C#) Create API should cast float default value

chobie opened the pull request:
I've found C# type issue on generated code.

defaultvalue.fbs
```
table FloatDefaultValue {
  value:float = 1.0;
}
```

generated create method like this:

```
public static Offset<FloatDefaultValue> CreateFloatDefaultValue(FlatBufferBuilder builder,
      float value = 1.0) {
    //  float value = 1.0 is syntax error
    //  it should be
    // `float value = (float)1.0` or `float value = 1.0f`
    builder.StartObject(1);
    FloatDefaultValue.AddValue(builder, value);
    return FloatDefaultValue.EndFloatDefaultValue(builder);
  }
```

I think to modify `GenDefaultValueBasic` function is much better than current PR'ed code. 
but it has slightly bigger effect: (at least should check getter properties due to remove duplicate (float)cast) 

do you have any good ideas?

evolutional commented:
I think I'd prefer if the C# code emitted the correct float value ```1.0f``` rather than ```(float)1.0```

macden1213456 commented:
=Call ,,”“+1-855-864-6325::;;// ((( Norton customer support phone number

=Call ,,”“+1-855-864-6325::;;// ((( Norton customer support phone number=Call ,,”“+1-855-864-6325::;;// ((( Norton customer support phone number=Call ,,”“+1-855-864-6325::;;// ((( Norton customer support phone number=Call ,,”“+1-855-864-6325::;;// ((( Norton customer support phone number=Call ,,”“+1-855-864-6325::;;// ((( Norton customer support phone number

norton 360phone number #$#1-855-864-6325##$norton 360Internet security phone number norton 360security phone number norton 360Internet security phone number #$#1-855-864-6325##$norton 360support phone number norton 360symantec phone number norton 360phone number customer service #$#1-855-864-6325##$norton 360customer service phone number norton 360tech support phone number norton 360technical support phone number norton 360phone numbers norton 360help phone number norton 360Internet security support phone number #$#1-855-864-6325##$norton 360phone support norton 360contact phone number norton 360internet security phone number norton 360Internet security support phone number norton 360phone norton 360800 phone number phone number for norton 360phone numbers norton 360Internet security customer service phone number norton 360customer support phone number find a phone number norton 360uk phone number usa phone number phone number for norton 360Internet security tech support phone number norton 360Internet security phone support free phone numbers phone number norton 360phone number get human find phone number phone number for norton 360security us phone number phone number usa #$#1-855-864-6325##$norton 360Internet security contact phone number norton 360phone number for support norton 360contact number phone number for norton 360Internet security usa phone numbers phone number search us phone numbers find phone numbers norton 360helpline phone number business phone numbers residential phone numbers irs phone number phone number directory free phone number norton 360number norton 360whose phone number is this norton 360Internet security norton 360account norton 360Internet security find a phone number free norton 360internet security norton 360support number norton 360security norton 360phone number billing symantec norton 360removal tool norton 360support norton 360support phone number usa norton 360renewal phone number #$#1-855-864-6325##$symantec phone number norton 360utilities We are also a tech support provider for third party products, the difference between us and other organizations are, we have engineers who have worked with tech-giants and Antivirus companies, we understand the product better than any other organizations, we have been in this service industry for more than a decade now, so a huge experience, great knowledge and utmost dedication are our strength. A service call is the most important thing that we have here, so a call is just what a user need to do, rest is our work. Norton Antivirus tech support number is now available for all its users across the globe.norton 360download norton 360virus phone number my norton 360account norton 360Internet security download norton 360customer service norton 360anti virus norton 360account login norton 360sign in norton 360australia phone number norton 360canada phone number norton 360symantec Internet security norton 360nortons#$#1-855-864-6325##$ norton 360login tech support phone number download norton 360information phone number norton 360my account symantec support anti virus get phone number norton 360virus norton 360Internet security support norton 360sales phone number norton 360by symantec phone number my norton 360norton 360internet security download norton 360trial norton 360virus protection norton 360Internet security technical support phone number norton 360Internet security customer service phone number norton 360telephone number norton 360security suite norton 360contact norton 360Internet security trial norton 360phone support number norton 360Internet security technical support phone number symantec norton 360norton 360help norton 360products norton 360online norton 360internet phone numbers search norton 360free trial norton 360tech support norton 360phone number uk norton 360security download contact norton 360norton 360renewal phone numbers usa norton 360by symantec norton 360update norton 360internet security customer service phone number #$#1-855-864-6325##$my norton 360account login norton 360Internet security phone support number download norton 360Internet security norton 360product key home phone numbers norton 360Internet security customer support phone number nortons Internet security norton 360Internet security product key norton 360power eraser nortan norton 360ghost anti virus protection reinstall norton 360norton 360canada norton 360Internet security phone norton 360key install norton 360symantec support phone number norton 360phone number customer support norton 360website check norton 360free download norto norton 360customer service number norten norton 360Internet security update free norton 360norton 360software norton 360ghost l5 norton 360backup norton 360phone number support norton 360Internet security phone number tech support norton 360billing phone number norton 360uk norton 360technical support mcfee norton 360firewall norton 360internet security product key norton 360anti virus norton 360live update norton 360Internet security customer service norton 360customer service phone number usa buy norton 360get a phone number norton 360tech support number virus protection #$#1-855-864-6325##$norton 360customer support phone number for norton 360support norton 360Internet security support norton 360protection norton 360Internet security review norton 360website download norton 360internet security norton 360account sign in norton 360Internet security Internet security norton 360uninstaller free norton 360Internet security norton 360Internet security Internet security norton 360Internet security phone number support northon norton 360helpline norton 360Internet security login how to get phone numbers nortonInternet security norton 360Internet security trial norton 360Internet security number norton 360log in norton 360Internet security free phone numbers free norton 360disk doctor norton 360systemworks norton 360Internet security download internet security norton 360virus removal norton 360account phone number norton 360live chat buy norton 360Internet security norton 360internet security login norton 360search norton 360software phone number anti virus software norton 360Internet security tech support phone number free norton 360Internet security norton 360mobile norton 360free norton 360Internet security software norton 360free Internet security norton 360Internet security phone number customer service norton 360online backup norton 360Internet security phone support notron quick heal norton 360toll free number #$#1-855-864-6325##$norton 360download manager norton 360identity safe download norton 360Internet security norton 360install norton 360internet security 20ll norton 360anti virus phone number phone number for norton 360customer service norton 360customer phone number norton 360customer support phone number canada symantec removal tool norton 360safe search norton 360Internet security price norton 360toolbar norton 360av norton 360Internet security coupon norton 360nz phone number norton 360800 number norton 360downloads norton 360Internet security free trial norton 360internet security 20l2 norton 360cancellation phone number norton 360updates phone book phone numbers in usa telephone number norton 360helpline number phone directory contact norton 360by phone norton 360subscription norton 360ghost download norton 360com support phone number norton 360Internet security contact norton 360Internet security contact number technical support phone number norton 360phone Internet security norton 360Internet security free norton 360support phone number us norton 360Internet security removal tool phone number for norton 360Internet security customer service norton 360Internet security update norton 360Internet security number norton 360Internet security phone number canada norton 360phone support canada norton 360spyware toll free numbers norton 360store norton 360online family find free phone numbers norton 360Internet security account Internet security norton 360gratis norton 360Internet security account norton 360Internet security contact phone number norton 360system works norton 360internet security support phone number symantec #$#1-855-864-6325##$norton 360Internet security norton 360utilities l5 contact norton 360Internet security nortons Internet security norton 360Internet security renewal norton 360Internet security support phone number usa norton 360customer service phone number billing norton 360internet security support usa number norton 360review norton 360annual renewal free Internet security norton 360anti virus norton 360norton 360Internet security renewal norton 360Internet security online trend Internet security how to contact norton 360Internet security by phone norton 360anitvirus norotn support norton 360norton 360security contact number us numbers norton 360Internet security phone support number norton 360technical support number norton 360customer care norton 360help desk phone number contact norton 360Internet security by phone norton 360Internet security tech support number norton 360Internet security tech support norton 360Internet security customer service norton 360anti virus software #$#1-855-864-6325##$norton 360anti virus protection search for phone numbers norton 360activation what is norton 360Internet security norton 360Internet security full norton 360support phone norton 360contact us norton 360renewal code Internet security norton 360free symantec technical support phone number norton 360security customer service phone number norton 360usa norton 360security support tech support number Internet security support install norton 360Internet security phone number for norton 360Internet security support live chat phone numbers norton 360Internet security customer service phone number usa phone number in us norton 360internet security 20l0 norton 360ghost l5.0 symantec tech support phone number norton 360Internet security customer support phone number vanity numbers phone number service norton 360my account login symantec technical support norton 360Internet security support number Internet security norton 360Internet security symantec support number norton 360for phone phone number of usa phone number for business norton 360l 800 number phone number for norton 360internet security norton 360hotline norton 360internet security contact phone number symantec #$#1-855-864-6325##$norton 360Internet security phone number for norton 360tech support norton 360customer support phone number usa norton 360Internet security support symantec contact norton 360security symantec customer service phone number norton 360Internet security 20ll norton 360customer service phone number uk norton 360Internet security contact number how to remove norton 360norton 360customer service phone numbers norton 36020ll norton 360Internet security help norton 360customer care phone number norton 360online support helpline phone number l 800 phone numbers norton 360sonar norton 360security phone support norton 360security contact Internet security norton 360contact norton 360Internet security customer service phone number information phone numbers norton 360Internet security sign in norton 360Internet security price norton 360customer service telephone number www norton 360com #$#1-855-864-6325##$ support phone numbers antiviry norton 360Internet security help desk phone number norton 360subscription renewal norton 360help desk l 800 number what is norton 360norton 360internet security contact number nortan Internet security Internet security free norton 360norton 360Internet security customer service number norton 360pc checkup how to contact norton 360Internet security by phone norton 360security contact phone number phone numbers us contact norton 360internet security norton 360Internet security full how do i find a phone number for free norton 360contact number uk norton 360internet security renewal cheap norton 360Internet security norton 360deals norton 360customer service number usa norton 360phone contact norton 360customer care number norton 360Internet security help phone number for norton 360Internet security norton 36020l2 #$#1-855-864-6325##$norton 360Internet security help phone number support for norton 360norton 360help number call norton 360norton 36020l0 norton 360security customer service how do i find phone numbers norton 360Internet security customer service phone number usa norton 360l800 number norton 360Internet security for phone install norton 360internet security norton 360symantec customer service phone number norton 360Internet security sale phone number for norton 360symantec customer service norton 360l norton 360sale norton 360support uk phone number nortion norton 360Internet security canada norton 360installation symantec customer service norton 360Internet security 800 number how to contact norton 360by phone norton 360home compare norton 360products norton 360customer support number anti virus products norton 360problems norton.com phone number norton 360800 find contact number norton 360siteadvisor #$#1-855-864-6325##$norton 360Internet security telephone number norton 360Internet security account login business phone numbers free norton 360Internet security deals norton 360symantec support phone number norton 360Internet security updates contact norton 360Internet security norton 360price nortan anti virus symantec customer support phone number how to install norton 360Internet security norton 360site norton 360an contact norton 360support download norton 360anti virus norton 360reviews norton 360Internet security 20l0 norton 360Internet security protection phone number for norton 360by symantec norton 360Internet security 5.0 phone numbers in us get a phone number for free symantec tech support norton 360Internet security support number norton 360helpdesk #$#1-855-864-6325##$norton 360sales norton 360Internet security technical support number norton 360customer service phone norton 360Internet security phone norton 360internet security help norton 360live update not working norton 360Internet security toll free number norton 360customer support telephone number norton 360Internet security canada contact phone number norton 360email norton 360login to my account norton 360security help norton 360customer service number us norton 360Internet security telephone number how to get phone numbers free phone number for norton 360Internet security billing symantec help vanity toll free numbers norton 360support website norton 360Internet security technical support phone numbers of norton 360Internet security v5 phone number norton 360Internet security customer service norton 360symantec support #$#1-855-864-6325##$contact norton 360Internet security by phone symantec contact norton 360homepage norton 360Internet security customer service phone number us norton 360Internet security customer support number symantec contact number find a free phone number symantec help desk phone number how to install norton 360norton 360Internet security phone contact telephone support by phone numbers norton 360a norton 360Internet security customer support norton 360internet security 20ll download norton 360service Internet security tech support buy norton 360direct renewal phone number business norton 360Internet security 3 user norton 360online help get phone numbers for free technical support number norton 360tech support telephone number anti virus anti virus norton 360services phone number for symantec customer service symantec phone number customer service norton 360security telephone number norton 360Internet security canada information on phone number norton 360customer services phone number phone number of us norton 360internet security contact norton 360antivrus best Internet security for phone norton 360internet security telephone number get phone number for free norton 360live norton 360support telephone number norton 360customer services norton 360for free best norton 360Internet security usa number phone norton 360email address symantec customer support phone numbers of usa vanity 800 numbers norton 360billing support for Internet security phone numbers in the us how much is norton 360norton 360contact number us norton 360Internet security contact contact number for norton 360support for norton 360Internet security norton 360Internet security symantec norton 360contact number usa find free phone number norton 360uk contact norton 360telephone number customer service norton 360server Internet security norton 360Internet security helpline number norton 360Internet security tech support norton 360internet security customer service norton 360cleansweep norton 360customer service phone number us symantec technical support number norton 360Internet security customer service number latest norton 360Internet security norton 360Internet security help and support phone number email tech support anti virus full toll free vanity numbers norton 360contact telephone number Internet security technical support norton 360contact number for customer service anti virus service tech support usa norton 360Internet security 4.0 usa contact number where to find phone numbers telephone number for norton 360how to contact norton 360norton 360telephone support norton 360help desk number norton 360Internet security usa norton 360Internet security symantec norton 360student norton 3603 user norton 360search engine how do you find phone numbers email technical support norton 360Internet security customer support norton 360support number usa norton 360Internet security problems tech support for norton 360norton 360Internet security website norton 360contact numbers norton 360customer service canada help me find a phone number norton 360Internet security account login norton 360live support phone with numbers norton 360Internet security helpline norton 360Internet security products information for phone numbers norton 360contact number for billing norton 360virus software chat with phone number norton 360repair norton 360Internet security usa anti virus a contact number for norton 360Internet security call norton 360support renew norton 360Internet security norton 360Internet security version 5.0 norton 360customer service toll free number norton 360official website 800 vanity numbers phone numbers business free phone number for business norton 360corporate usa tech support norton 360Internet security helpline norton 360troubleshooting norton 360confidential get free phone numbers where can i get phone numbers for free about norton 360Internet security Internet security products phone number in the us phone contact numbers cheap norton 360call norton 360Internet security nortons anti virus toll free phone support phone numbers #$#1-855-864-6325##$symantec customer support center problems with norton 360anti virus business help phone numbers norton 360Internet security Internet security free vanity toll free phone numbers symantec phone support information on a phone number norton 360antiviris norton 360anit virus norton 360customer service 800 number phone norton 360Internet security free business phone numbers norton 360Internet security technical support phone tech support anti norton 360virus norton 360save and restore contact symantec support virus protection norton 360customer service number for norton 360Internet security phone numbers find norton 360internet Internet security norton 360telephone an virus symantec customer service number free information for phone numbers how can i find phone numbers norton 360toll free number usa #$#1-855-864-6325##$norton 360Internet securitytm find an phone number phone service number norton 360Internet security support telephone number norton 360online Internet security what this phone number how to contact norton 360Internet security norton 360contact information norton 360Internet security contact information norton 360Internet security customer service phone norton 360Internet security subscription norton 360symantec customer service customer services number norton 360help support norton 360Internet security tech support number phone numbers in the usa tech support numbers vanity 800 number norton 360Internet security for free technical support for norton 360telephone number for norton 360Internet security norton 360Internet security 2OO9 toll free 800 norton 360Internet security removal norton 360and norton 360nortons support personalized phone numbers norton 360toll free Internet security help call #$#1-855-864-6325##$norton 360Internet security support free Internet security norton 360Internet security Internet security norton 360Internet security symantec tech support number tech support in usa is norton 360free norton 360Internet security us how phone number symantec customer support number 800 phone email anti-virus norton 360technical support number usa how do i get phone numbers for free nortonanti virus norton 360Internet security server norton 360netbook edition contact technical support free tech support phone number phone number l 800 what anti virus business tech support free number usa norton 360canada support symantec business support norton 360from symantec norton 360live chat support norton 360antispyware a free phone number email norton 360norton 360for business customer service norton 360cheapest norton 360norton 360faq free contact information call technical support toll free 800 numbers norton 360contact us number norton 360support usa anti virus on phone norton 360Internet security telephone support norton 360Internet security brasil get norton 360free norton 360Internet security Internet security norton 360Internet security information symantec telephone number email support number norton 360live help free toll number symantec help desk norton 360support canada norton 360Internet security contact us norton 360anti virus update number for norton 360Internet security norton 360create account norton 360contact telephone norton 360virus list norton 360support email login norton 360Internet security customer service number for norton 360norton 360help and support norton 360support toll free number phone norton 360norton 360contact customer support norton 360customer 800 number for norton 360Internet security customer service telephone number norton 360Internet security by symantec norton 360Internet security business best norton 360anti vyras norton 360virus help norton 360support.com norton 360Internet security by symantec #$#1-855-864-6325##$norton 360helpline uk 800 vanity number contact help number telephone number for norton 360Internet security customer service norton 360Internet security live chat what are anti virus contact tech support norton 360Internet security products#$#1-855-864-6325##$###########@

gwvo commented:
What about this commit? It seems to be missing the generated code.

chobie commented:
@evolutional thanks pointing the issue! I've missed that.

@gwvo ah, sorry, I went sleep before describe the commit. first commit added `(float)` cast. then second commit removed type cast and fixed correct float value (also, fixed related code which uses `GenDefaultValue`).

>  It seems to be missing the generated code.

Current test code haven't used `table` default value for float type. so its no effect for existing generated code. (note: struct seems not use default value.)

I'll do more check in this week and update comment.

gwvo commented:
ok, let me know.

chobie commented:
https://gist.github.com/chobie/61fab9086081be6b76c8

Sorry for late update. probably it is fine. should I add field to `monster_test.fbs`? 
(for example, luck:float = 1 (id:25);)

gwvo commented:
Yes, would be good to add such a field. Also regenerate code for other languages when you do that, and run tests if you can.

gwvo commented:
status?

chobie commented:
oops, sorry. I'm having a hectic half a year.
I'll to this in this weekend.

gwvo commented:
Thanks!

chobie commented:
I've checked FloatDefaultValue table with `d70f5ac6b02f250dca8e2c6e8f59a4223d1f66f6`.
this issue has been fixed. sorry about the mess here.

generated c# file is below:

```
// automatically generated by the FlatBuffers compiler, do not modify

using System;
using FlatBuffers;

public sealed class FloatDefaultValue : Table {
  public static FloatDefaultValue GetRootAsFloatDefaultValue(ByteBuffer _bb) { return GetRootAsFloatDefaultValue(_bb, new FloatDefaultValue()); }
  public static FloatDefaultValue GetRootAsFloatDefaultValue(ByteBuffer _bb, FloatDefaultValue obj) { return (obj.__init(_bb.GetInt(_bb.Position) + _bb.Position, _bb)); }
  public FloatDefaultValue __init(int _i, ByteBuffer _bb) { bb_pos = _i; bb = _bb; return this; }

  public float Value { get { int o = __offset(4); return o != 0 ? bb.GetFloat(o + bb_pos) : (float)1.0f; } }

  public static Offset<FloatDefaultValue> CreateFloatDefaultValue(FlatBufferBuilder builder,
      float value = 1.0f) {
    builder.StartObject(1);
    FloatDefaultValue.AddValue(builder, value);
    return FloatDefaultValue.EndFloatDefaultValue(builder);
  }

  public static void StartFloatDefaultValue(FlatBufferBuilder builder) { builder.StartObject(1); }
  public static void AddValue(FlatBufferBuilder builder, float value) { builder.AddFloat(0, value, 1.0f); }
  public static Offset<FloatDefaultValue> EndFloatDefaultValue(FlatBufferBuilder builder) {
    int o = builder.EndObject();
    return new Offset<FloatDefaultValue>(o);
  }
};

````

I'm closing this issue. thanks
